### ISSUE TITLE
Training details of adding visual vocabs.

### ISSUE THREAD

**Issue by martian422** [jy0205/LaVIT#20]
Thanks for your excellent work :)
We admire the potential of vision-language models with unified vocabularies, and wish if some templates of the training code can be released for academic experiments.
As inserting a large amount of vocabs into LLM and initialize them from scratch may lead to instability, can you tell us or hint the strategy details you applied when adding these 16k new visual vocabs into the llama model in stage 2?

**Comment by jy0205** [comment]
In our implementation, the added 16k new visual vocabs are randomly initialized in the same way as the original word vocab in llama2. When joint training, using a small learning rate for the LLM part is very important. We suggest the learning rate of e−5 level for LLM and e−4 level for visual part (Not the visual vocab). Besides, text corpus data is also important. We suggest a proportion of 2:1 (L:VL) to maintain its original learned knowledge.
